components/policies/staging/base/konflux-rbac/set-tenant-namespace-visibility/.chainsaw-test/chainsaw-test.yaml

```
@@ -0,0 +1,221 @@
+# These tests assume that a tenant namespace already has the `konflux-ci.dev/visibility=private`
+# label and does not apply to the following scenarios:
+# 1. A non-tenant namespace is changed into a private tenant namespace (no konflux-ci.dev/visibility
+#    label will be added).
+# 2. A private tenant namespace is changed into a non-tenant namespace (the
+#    konflux-ci.dev/visibility label will not be removed).
+
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: set-tenant-namespace-visibility-to-public
+spec:
+  description: |
+    tests that a Namespace labeled with `konflux-ci.dev/type=tenant`
+    will also get labeled with `konflux-ci.dev/visibility=public` if
+    there is a RoleBinding for the `system:authenticated group` in the
+    same Namespace
+  bindings:
+  - name: type
+    value: tenant
+  steps:
+  - name: given-kyverno-has-permission-on-resources
+    try:
+    - apply:
+        file: ../kyverno_rbac.yaml
+  - name: apply-set-tenant-namespace-visibility-clusterpolicy
+    try:
+    - apply:
+        file: ../set-tenant-namespace-visibility-clusterpolicy.yaml
+  - name: when-tenant-namespace-exists
+    try:
+    - apply:
+        file: ./resources/tenant-namespace.yaml
+  - name: when-authenticated-rolebinding-exists
+    try:
+    - apply:
+        file: ./resources/authorized-rolebinding.yaml  
+  - name: then-visibility-is-public
+    try:
+    - assert:
+        file: ./resources/expected-public-tenant-namespace.yaml
+
+---
+
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+   name: do-not-modify-non-tenant-namespace
+spec:
+  description: |
+    tests that a non-tenant Namespace (one without a 
+    `konflux-ci.dev/type=tenant` label) will not get get labeled 
+    with a `konflux-ci.dev/visibility` label if a RoleBinding
+    for the `system:authenticated group` is created
+  bindings:
+  - name: type
+    value: non-tenant
+  steps:
+  - name: given-kyverno-has-permission-on-resources
+    try:
+    - apply:
+        file: ../kyverno_rbac.yaml
+  - name: apply-set-tenant-namespace-visibility-clusterpolicy
+    try:
+    - apply:
+        file: ../set-tenant-namespace-visibility-clusterpolicy.yaml
+  - name: when-non-tenant-namespace-exists
+    try:
+    - apply:
+        file: ./resources/non-tenant-namespace.yaml
+    - sleep:
+        duration: 5s
+  - name: when-authenticated-rolebinding-exists
+    try:
+    - apply:
+        file: ./resources/authorized-rolebinding.yaml  
```

we should not rely on sleep. If required we can increase timeouts
```suggestion
  - name: when-authenticated-rolebinding-exists
    timeout: 60s
    try:
    - apply:
        file: ./resources/authorized-rolebinding.yaml  
```